### PR TITLE
Extendable configurations.

### PR DESCRIPTION
**Pull request by izaakschroeder**
Refactor the single generator into several, allowing users to pick the configurations that best suit their setup. Eventually it should be possible to use this setup to generate eslint rule modules as well, for users with several project who want consistent linting across all of them.

This also updates the default from tabs to spaces.

### CODE REVIEW

File: lib/eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			return uniq(a.concat(b));
+		}
+	});
+}
+
+function update(dst, src) {
+	return json.update(dst, combine(json.parse(dst), src), { });
+}
+
+
```
Comment (jjt) [inline lib/eslintrc.js]: Any reason for the double blank like?

File: generators/filenames/templates/.eslintrc.json5 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+	"plugins": [
+		"filenames"
+	],
+
+	"rules": {
+		// Prevent people from naming their files in formats other than kebab-case.
+		// Rubyists have a bad habit of doing_this_for_js_files and others feel like
+		// variable naming conventionShouldMatchFiles. They're both wrong.
```
Comment (jjt) [inline generators/filenames/templates/.eslintrc.json5]: Maybe phrase this in a little more friendly/positive way?

**Comment by jjt** [comment]
Minor line notes, indentation, then LGTM!

File: lib/eslintrc.js (hunk 1 above)
Comment (izaakschroeder) [inline lib/eslintrc.js]: Derp.

**Comment by jjt** [comment]
:+1:

**Pull request merged**